core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1356,6 +1361,35 @@ public Expression implementResult(AggContext info,
     }
   }
 
+  /** Implementor for the {@code BIT_AND} and {@code BIT_OR} aggregate function. */
+  static class BitOpImplementor extends StrictAggImplementor {
+    @Override protected void implementNotNullReset(AggContext info,
+        AggResetContext reset) {
+      Object initValue = info.aggregation() == BIT_AND ? -1 : 0;
```

Comment:
Why is this -1 instead of 1?

Reply:
If it is bit and, we need to give it initial value 0xFFFFFFFF (e.g. int), which is -1.